### DISCUSSION TITLE
How to setup Google Analytics

### DISCUSSION THREAD

**Discussion by danielvlla** [nuxt/scripts#76] [Q&A]
Using the documentation, I installed nuxt/scripts succesfully for Google Analytics use by adding this to my Nuxt Config.

```
  scripts: {
    registry: {
      googleAnalytics: {
        id: 'MY_GOOGLE_ANALYTICS_ID'
      }
    }
  },
```

This is my first time implementing Google Analytics on a Nuxt site. I just want simple analytics (no need for custom events).

From what I understand, I need to implement a consent form for accepting or declining cookies right? This seems to be required especially in the EU. But then I am not sure what do to with that information and how to communicate it to Google Analytics using nuxt/scripts.

Am I missing anything else please?

**Comment by harlan-zw** [comment]
Hi @danielvlla 

If you intend to use a consent form, you will need to implement google analytics using the composable in conjunction with [useScriptTriggerConsent](https://scripts.nuxt.com/docs/api/use-script-trigger-consent)

You could put this, for example, in your app.vue file.

```ts
const agreedToCookiesScriptConsent = useScriptTriggerConsent()

useScriptGoogleAnalytics({
   scriptOptions: {
     trigger: agreedToCookiesScriptConsent
  }
})
```

You will need to manually implement the UI yourself.

**Reply by madc** [reply]
Looking at the [docs for script triggers](https://scripts.nuxt.com/docs/guides/script-triggers) `trigger`must be nested in `scriptOptions` for registry scripts:

```
useScriptMetaPixel({
  id: '1234567890',
  scriptOptions: {
    // load however you like!
    trigger: new Promise(resolve => setTimeout(resolve, 3000))
  }
})
```

**Reply by harlan-zw** [reply]
Thanks, have updated my op
